### ISSUE TITLE
auto increment is not working for large number of rows while inserting...

### ISSUE THREAD

**Issue by Gayatri31** [balderdashy/sails-mongo#197]
hey i have a problem with auto incrementing... i need a field sequence_num as auto incrementing while insertion irrespective of _id. so i tried the Counter collection incrementing part and again adding to my collection , its working for only 100 to 150 documents at a time insertion but if i increase documents size as upto 200 or more it throwing me an ERROR as Failed to MongoDb Connection and even a single record inserting..., i'm not able to get to know that why this is happening... As of my insertion is as file data,if a file consisting of 100 rows my sequenc_num is incrementing but if 200 rows as file it was throwing the error...
My code is below can any one help to how to sort out this....

beforeCreate: function (values, cb) {
    Contacts.getNextSequence("sequence_num",function(obj){
      console.log(values.name);
      _.extend(values,{sequence_num:obj.sno});
      cb();
    });
  },

  getNextSequence :function(name,cb){
    var db = require("mongojs").connect('omerta', ['counters']);
    
    db.collection('counters').find(function(err,list){
      if(err) console.log(err);
      console.log(list);
    });

    db.collection('counters').findAndModify(
    {
      query: { _id: name },
      update: { $inc: { seq: 1 } } ,
      new: true,
      upsert: true,
      multi:true
    },
    function(err,inc, lastErrorObject){
      if(err) console.log(err);
      console.log(lastErrorObject);
      console.log(inc.seq);
      cb({sno:inc.seq});
    });
  },
and my counter mdel  is ...

/Counter.js model
module.exports = {
    attributes: {
        seq: 'Integer',
        _id: 'String'
    }
};

**Comment by sailsbot** [comment]
Thanks for posting, @Gayatri31.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
